diff -Nurd socket++-1.10/sockinet.C socket++-1.10-fixed/sockinet.C
--- socket++-1.10/sockinet.C	Wed Oct 18 01:07:32 1995
+++ socket++-1.10-fixed/sockinet.C	Tue May 30 12:23:19 2000
@@ -143,7 +143,7 @@
 sockinetaddr sockinetbuf::localaddr() const
 {
   sockinetaddr sin;
-  int len = sin.size();
+  size_t len = sin.size();
   if (::getsockname(rep->sock, sin.addr (), &len) == -1)
     perror("sockinetbuf::localaddr()");
   return sin;
@@ -167,7 +167,7 @@
 sockinetaddr sockinetbuf::peeraddr() const
 {
   sockinetaddr sin;
-  int len = sin.size();
+  size_t len = sin.size();
   if (::getpeername(rep->sock, sin.addr (), &len) == -1)
     perror("sockinetbuf::peeraddr()");
   return sin;
Binary files socket++-1.10/sockinet.o and socket++-1.10-fixed/sockinet.o differ
diff -Nurd socket++-1.10/sockstream.C socket++-1.10-fixed/sockstream.C
--- socket++-1.10/sockstream.C	Wed Oct 18 01:07:32 1995
+++ socket++-1.10-fixed/sockstream.C	Tue May 30 12:22:29 2000
@@ -273,7 +273,7 @@
 
 sockbuf	sockbuf::accept (sockAddr& sa)
 {
-  int len = sa.size ();
+  size_t len = sa.size ();
   int soc = -1;
   while ((soc = ::accept (rep->sock, sa.addr (), &len)) == -1
 	 && errno == EINTR)
@@ -319,7 +319,7 @@
   if (rtmo != -1 && is_readready (rtmo)==0) return 0;
   
   int	rval;
-  int	sa_len = sa.size ();
+  size_t	sa_len = sa.size ();
   
   if ((rval = ::recvfrom (rep->sock, (char*) buf, len,
 			  msgf, sa.addr (), &sa_len)) == -1)
@@ -490,7 +490,7 @@
 
 int sockbuf::getopt (option op, void* buf, int len, level l) const
 {
-  int	rlen = len;
+  size_t	rlen = len;
   if (::getsockopt (rep->sock, l, op, (char*) buf, &rlen) == -1)
     perror ("sockbuf::getopt");
   return rlen;
