--- Source/LibJXR/image/decode/segdec.c	2020-11-02 14:17:49.000000000 -0500
+++ Source/LibJXR/image/decode/segdec.c	2020-11-02 14:18:24.000000000 -0500
@@ -52,6 +52,16 @@
 //================================================================
 // Memory access functions
 //================================================================
+static U32 _byteswap_ulong(U32 bits)
+{
+    U32 r = (bits & 0xffu) << 24;
+    r |= (bits << 8) & 0xff0000u;
+    r |= ((bits >> 8) & 0xff00u);
+    r |= ((bits >> 24) & 0xffu);
+
+    return r;
+}
+
 static U32 _FORCEINLINE _load4(void* pv)
 {
 #ifdef _BIG__ENDIAN_
--- Source/LibJXR/jxrgluelib/JXRGlueJxr.c	2020-11-02 14:07:19.000000000 -0500
+++ Source/LibJXR/jxrgluelib/JXRGlueJxr.c	2020-11-02 14:20:18.000000000 -0500
@@ -28,6 +28,7 @@
 //*@@@---@@@@******************************************************************
 #include <limits.h>
 #include <JXRGlue.h>
+#include <wchar.h>
 
 
 static const char szHDPhotoFormat[] = "<dc:format>image/vnd.ms-photo</dc:format>";
